src/jit/emitxarch.cpp

```diff
@@ -12551,6 +12545,8 @@ size_t emitter::emitOutputInstr(insGroup* ig, instrDesc* id, BYTE** dp)
             assert(IsAVXInstruction(ins));
 
             code = insCodeRM(ins);
+            code = AddVexPrefixIfNeeded(ins, code, size);
+            code = insEncodeReg3456(ins, id->idReg2(), size, code); // encode source operand reg in 'vvvv' bits in 1's complement form
```

Comment:
So we unconditionally encode reg2 in VEX.vvvv field and special guys (may need to encode other operand in VEX.vvvv) will be handled in codegen or other places. Right?

Reply:
Yes, that is what we were already doing (for one of the code paths).

If the instruction has special encoding requirements, we should probably assert that they don't call `insEncodeReg3456` and are instead handled elsewhere.

Do you have a list of which instructions meet this requirement?